cmd/run/run.go

```diff
@@ -482,23 +465,41 @@ func RunServer(ctx context.Context, config *Config) error {
 
 	unaryInterceptors := []grpc.UnaryServerInterceptor{
 		grpc_validator.UnaryServerInterceptor(),
-		otelgrpc.UnaryServerInterceptor(otelgrpc.WithTracerProvider(tp)),
 		middleware.NewRequestIDInterceptor(logger),
-		middleware.NewStoreIDInterceptor(),
-		middleware.NewLoggingInterceptor(logger),
-		grpc_auth.UnaryServerInterceptor(middleware.AuthFunc(authenticator)),
 	}
 
 	streamingInterceptors := []grpc.StreamServerInterceptor{
 		grpc_validator.StreamServerInterceptor(),
-		otelgrpc.StreamServerInterceptor(otelgrpc.WithTracerProvider(tp)),
-		grpc_auth.StreamServerInterceptor(middleware.AuthFunc(authenticator)),
 		middleware.NewStreamingRequestIDInterceptor(logger),
+	}
+
+	if config.Metrics.Enabled {
+		unaryInterceptors = append(unaryInterceptors, grpc_prometheus.UnaryServerInterceptor)
+		streamingInterceptors = append(streamingInterceptors, grpc_prometheus.StreamServerInterceptor)
+
+		if config.Metrics.EnableRPCHistograms {
+			grpc_prometheus.EnableHandlingTimeHistogram()
+		}
+	}
+
+	if config.Trace.Enabled {
+		unaryInterceptors = append(unaryInterceptors, otelgrpc.UnaryServerInterceptor(otelgrpc.WithTracerProvider(tp)))
+		streamingInterceptors = append(streamingInterceptors, otelgrpc.StreamServerInterceptor(otelgrpc.WithTracerProvider(tp)))
```

Comment:
Can you drop the `otelgrpc.WithTracerProvider(tp)`? If it is not supplied it uses the global tracer provider, so this is redundant.

Reply:
Yep, good call. Done.